### PR TITLE
Update to turborepo 1.4

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Split CI and cypress workflows.
Setup remote caching in ci workflow.
Add inputs to store#codegen pipeline.

## Justify why they are needed

I split up the gh workflows since it should be enough to run e2e tests on every push to main branch (not every PR).

I updated the ci workflow according to: https://turborepo.org/docs/ci/github-actions

I added turborepo inputs according to new features described here: https://turborepo.org/blog/turbo-1-3-0

I setup remote caching to speed up our CI pipeline.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] Will this be part of a product update? If yes, please write one phrase about this update.

[GRW-1315]: https://hedvig.atlassian.net/browse/GRW-1315?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #350** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/350" target="_blank">Graphite</a> 
    * **PR #352** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/352" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/352?utm_source=stack-comment).

File: .github/workflows/cypress.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Cypress
+
+on:
+  push:
+    branches:
+      - main
```
Comment (alebedev) [inline .github/workflows/cypress.yml:6]: Do we want to run on review branches as well?

**Review by alebedev — APPROVED** [review]
Looks great!

File: .github/workflows/cypress.yml (hunk 1 above)
Comment (robinandeer) [reply .github/workflows/cypress.yml:6]: I discussed this with @guilhermespopolin a little and let's evaluate

But until then I think we can keep running only on production deploys

**Comment by robinandeer** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/352?utm_source=stack-comment).


Job Reference: 97Ni5h6J9snBETZbSL6j

**Pull request merged**